File: scripts/ts_scripts/bt_to_s4.ts
// bt_to_s4.ts
const processStands = (inputData: any, merchantId: string) => {
    // Extract stands and create a new list with the desired fields
    const stands = inputData.stands || [];
    const processedStands: any[] = [];
  
    for (const stand of stands) {
      const processedStand = {
        id: stand.id,
        uuid: stand.uuid,
        apple_pay_merchant_id: null,
        android_pay_merchant_id: null,
        merchant_account_id: null,
        merchant_id: merchantId,
      };
      processedStands.push(processedStand);
    }
  
    const outputData = {
      stands: processedStands,
    };
  
    return outputData;
  };
  
  export default processStands;
  

File: scripts/BRAINTREE_TO_SHIFT4/bt_to_s4.py
import json

def process_stands(input_file, output_file, merchant_id):
    # Load the input JSON file
    with open(input_file, 'r') as infile:
        data = json.load(infile)
    
    # Extract stands and create a new list with the desired fields
    stands = data.get('stands', [])
    processed_stands = []

    for stand in stands:
        processed_stand = {
            "id": stand.get("id"),
            "uuid": stand.get("uuid"),
            "apple_pay_merchant_id": None,
            "android_pay_merchant_id": None,
            "merchant_account_id": None,
            "merchant_id": merchant_id
        }
        processed_stands.append(processed_stand)

    # Create the output JSON structure
    output_data = {"stands": processed_stands}

    # Write the output JSON to a file
    with open(output_file, 'w') as outfile:
        json.dump(output_data, outfile, indent=4)

if __name__ == "__main__":
    # Input and output file paths
    input_file = "input.json"
    output_file = "output.json"

    # Ask the user for the merchant_id to apply to all stands
    merchant_id = input("Enter the merchant_id to apply to all stands: ")

    # Process the stands data and write to the output file
    process_stands(input_file, output_file, merchant_id)

    print(f"Processed data written to {output_file}")
